### PR TITLE
Model2 Approach to a Unified Common-Utils

### PR DESCRIPTION
**Pull request by okram**
This is related to ISSUE-10 of security analytics. It's the best push I have that uses a simpler Model infrastructure based on public fields and reflection. It also incorporates the notion of redirecting existing actions to a intermediate "translator action" (@getsaurabh02 idea). 

This is a 3 part PR across common-utils, alerting, and security analytics. All under the same `model2/` branch.

*CheckList:*
[X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by okram** [comment]
Here is a script that I wrote that will build the `opensearch-alerting-2.2.0-SNAPSHOT.zip`. The build order is important for these 3 repos. You have to do: `common-utils/`, then `alerting/`, then `security-analytics/`. Between each, make sure your local `.m2` Maven repo has the previously build `SNAPSHOT` or else you'll pull it from remote. And before you do `security-analytics/`, you have to copy over (and rename `.zip`) the `alerting/` jar.

Because of the MacOSX build via Docker, there are some `.m2/` antics that shouldn't be necessary via a pure Linux build.

```
#!/bin/bash

mkdir /docker/tmp/.m2
mkdir /docker/tmp/.m2/repository
mkdir /docker/tmp/.m2/repository/org/
pwd; pushd common-utils
gradle build publishToMavenLocal --info; popd
cp -rf ~/.m2/repository/org/opensearch/ /docker/tmp/.m2/repository/org/
pwd; pushd alerting
gradle test publishPluginZipPublicationToMavenLocal --info; popd
cp ~/.m2/repository/org/opensearch/opensearch-alerting/2.2.0.0-SNAPSHOT/opensearch-alerting-2.2.0.0-SNAPSHOT.jar security-analytics/src/test/resources/alerting/opensearch-alerting-2.2.0.0.zip
cp -rf ~/.m2/repository/org/opensearch/ /docker/tmp/.m2/repository/org/
pwd; pushd security-analytics
gradle build --info; popd
```

File: src/main/java/org/opensearch/securityanalytics/alerting/resthandlers/RestIndexMonitorAction.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        return channel -> {
+            LOG.info("{} {}/{}", request.method(), SAP_MONITORS_BASE_URI, getName());
+            final String monitorId = request.param(MONITOR_ID);
+            if (null == monitorId || monitorId.isEmpty()) throw new IllegalArgumentException("missing monitorID");
+            /*long seqNo = request.paramAsLong(IF_SEQ_NO, SequenceNumbers.UNASSIGNED_SEQ_NO);
+            long primaryTerm = request.paramAsLong(IF_PRIMARY_TERM, SequenceNumbers.UNASSIGNED_PRIMARY_TERM);
+            WriteRequest.RefreshPolicy refreshPolicy = (request.hasParam(REFRESH)) ? WriteRequest.RefreshPolicy.parse(request.param(REFRESH)) : WriteRequest.RefreshPolicy.IMMEDIATE;
+            XContentParser parser = request.contentParser();
+            final Monitor monitor = ModelSerializer.read(parser, Monitor.class);*/
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/securityanalytics/alerting/resthandlers/RestIndexMonitorAction.java:56]: nit:cleanup ?

File: src/main/java/org/opensearch/securityanalytics/alerting/resthandlers/RestIndexMonitorAction.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+            client.execute(
+                    IndexMonitorAction.SAP_ALERTING_BRIDGE_INSTANCE,
+                    new IndexMonitorRequest(monitorId, 3L, 4L, WriteRequest.RefreshPolicy.IMMEDIATE, request.method(), new Monitor("anid","atype",12L,"hello",235L,"seconds",List.of())),
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/securityanalytics/alerting/resthandlers/RestIndexMonitorAction.java:60]: Looks like a test code. Why aren't we constructing the transport fields from the rest requests

File: src/main/java/org/opensearch/securityanalytics/alerting/transport/TransportExecuteMonitorAction.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        super(ExecuteMonitorAction.ALERTING_NAME, actionFilters, transport.getTaskManager());
+        this.client = client;
+        this.cluster = clusterService;
+
+    }
+
+    @Override
+    protected void doExecute(final Task task, final ExecuteMonitorRequest request, final ActionListener<ExecuteMonitorResponse> actionListener) {
+        try {
+            actionListener.onResponse(new ExecuteMonitorResponse(new Monitor("dsaf", "Atype", 135L, "nae", 245L, "asdf", List.of())));
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/securityanalytics/alerting/transport/TransportExecuteMonitorAction.java:50]: Is this test code again?

File: src/test/java/org/opensearch/securityanalytics/alerting/ExampleAlertingJSON.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.securityanalytics.alerting;
+
+import org.json.JSONObject;
+
+public class ExampleAlertingJSON {
```
Comment (getsaurabh02) [inline src/test/java/org/opensearch/securityanalytics/alerting/ExampleAlertingJSON.java:10]: What is the use of this?

File: src/test/java/org/opensearch/securityanalytics/alerting/ExampleIndexMonitorJSON.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.securityanalytics.alerting;
+
+import org.json.JSONObject;
+
+public class ExampleIndexMonitorJSON {
+
+    public static String exampleIndexMonitor = "{\n" +
```
Comment (getsaurabh02) [inline src/test/java/org/opensearch/securityanalytics/alerting/ExampleIndexMonitorJSON.java:12]: Let's define these as resources (resource json files) instead?

File: src/main/java/org/opensearch/securityanalytics/alerting/resthandlers/RestExecuteMonitorAction.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.securityanalytics.Tokens.*;
+
+public class RestExecuteMonitorAction extends BaseRestHandler {
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/securityanalytics/alerting/resthandlers/RestExecuteMonitorAction.java:26]: Java docs across will be useful

**Review by getsaurabh02 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**